File: java/Patterns/src/com/bilak/structural/facade/subcomponents/PaymentService.java
package com.bilak.structural.facade.subcomponents;

import com.bilak.structural.facade.domain.Client;

import java.time.LocalDate;

/**
 * @author Yuliana
 * @version 1.0.0
 * @group 343(1)
 * @department PZKS
 * @project java
 * @class PaymentService
 * @since 10/14/2022 - 16.40
 **/
public class PaymentService {
    public static boolean makePayment(double priceToPay, Client client) {
        if (LocalDate.now().getDayOfWeek().getValue() >= 6) {
            priceToPay *= 1.05;
        }
        System.out.println("Withdrawing " + priceToPay + " UAH from client's " + client.getName() + " bank account...");
        return true;
    }
}

File: javascript/structural/facade/payment-service.js
export class PaymentService {
    makePayment(priceToPay, client) {
        const dayOfWeek = new Date().getDay();
        if (dayOfWeek === 0 || dayOfWeek === 6) {
            priceToPay *= 1.05;
        }
        console.log(`Withdrawing ${priceToPay} UAH from client's ${client.name} bank account...`);
        return true;
    }
}